storage/client/blob/src/main/java/com/azure/storage/blob/BlobClientBuilder.java

```diff
@@ -121,29 +129,77 @@ private AzureBlobStorageBuilder buildImpl() {
 
     /**
      * @return a {@link BlobClient} created from the configurations in this builder.
+     * @throws NullPointerException If {@code endpoint}, {@code containerName}, or {@code blobName} is {@code null}.
      */
-    public BlobClient buildClient() {
-        return new BlobClient(buildAsyncClient());
+    public BlobClient buildBlobClient() {
```

Comment:
Now that we build all four types of blob clients from the same builder, we should have the docstrings each have a little something of a main description talking about what the clients are for. Specifically, I want them to touch on how Blob(Async)Client is for generic blob methods like download and properties for when you don't know what kind of blob it is, and that the other three are for use when you know the type of blob you are interacting with.

Reply:
I agree!

Reply:
Added comments to explain when to build which client.